### PR TITLE
Enable build cache for jammy

### PR DESCRIPTION
**Pull request by svenevs**
AFAICT this is all we need to update to enable the build cache for jammy.  Requires

- [X] Create the jammy provisioned image (currently blocked, see offline chat).
- [X] Add a provisioned release job (or alternatively one of the regex selected jobs being changed here).
- [X] Launch a build using this branch and confirm the cache is getting used.
- [X] Test some focal jobs against this PR that use cache to make sure new name detection still works as expected in CI (not just locally).
    - [`linux-focal-gcc-bazel-experimental-release` #11398](https://drake-jenkins.csail.mit.edu/view/Linux%20Focal/job/linux-focal-gcc-bazel-experimental-release/11398/) took 4min 12sec.
    - [`linux-focal-gcc-bazel-experimental-debug` #1596](https://drake-jenkins.csail.mit.edu/view/Linux%20Focal/job/linux-focal-gcc-bazel-experimental-debug/1596/) took 6min 24sec.
- [X] Test everything debug / release jammy jobs.  These are not expected to benefit from the cache, the cache was populated by jobs with differing compilation definitions (will be resolved by future continuous / nightly).  Both jobs connected to / read from the cache, just not much speedup since it wasn't populated previously.
    - 51min 47sec: [`linux-jammy-gcc-bazel-experimental-everything-debug` #2](https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-everything-debug/2/console)
    - 45min 3sec: [`linux-jammy-gcc-bazel-experimental-everything-release` #2](https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-everything-release/2/console)

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/169)

### CODE REVIEW

**Review by svenevs — COMMENTED** [review]
+@BetsyMcPhail for review and/or potential handoff depending on the first two.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/169#-:-N6GbOZG6F3vPPmnnR06:bgxpp5n)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @BetsyMcPhail)

**Review by BetsyMcPhail — COMMENTED** [review]
First item is waiting for #170 

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/169#-:-N6JeG9o-KRQ--dWM-LY:b-xqy3lb)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @BetsyMcPhail)

**Comment by svenevs** [comment]
Status update.  Jammy cache seems to be working, some notes:

1. AFAIU the logic here

    https://github.com/RobotLocomotion/drake-ci/blob/9b42a3ad7d2064f194d06c7686905623ffcf6edf/driver/configurations/bazel.cmake#L96-L118

    means that continuous and release both ignore the existing cache (but upload the final results to it), whereas experimental only reads from the cache (but does not upload its results to it).  That seems fine / what we want, but for testing I had to introduce some debug commits (see next item).
2. In order to test the cache I temporarily allowed the experimental jammy release and debug builds to upload to the cache.
    - Release job that populated the cache: ~30 minutes, https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-release/7/
    - Release job that used this cache: 1min 40 sec, https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-release/8/
    - Debug job that populated the cache: 37 minutes, https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-debug/7/
    - Debug job that used the cache: 4 min 20sec, https://drake-jenkins.csail.mit.edu/view/Linux%20Jammy/job/linux-jammy-gcc-bazel-experimental-debug/8/
3. Given the changeset here I need to find and run some focal jobs against this PR that would use the cache to make sure they are still working.
4. Testing the everything-debug and everything-release jammy jobs against this PR is a good idea (there were some issues with the install script and I'm not 100% sure that the gurobi license is going to work).

File: driver/configurations/cache.cmake [hunk 1]
```diff
@@ -392,9 +392,11 @@ else()
       OUTPUT_VARIABLE DASHBOARD_LSB_RELEASE_DESCRIPTION_OUTPUT_VARIABLE
       OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    # NOTE: Focal reports `Ubuntu 20.04.4 LTS` whereas Jammy reports
+    # `Ubuntu 22.04 LTS`, the third number must be optional.
     if(DASHBOARD_LSB_RELEASE_DESCRIPTION_RESULT_VARIABLE EQUAL 0 AND
        DASHBOARD_LSB_RELEASE_DESCRIPTION_OUTPUT_VARIABLE MATCHES
-       "^Ubuntu ([0-9]+[.][0-9]+[.][0-9]+) LTS$"
+       "^Ubuntu ([0-9]+[.][0-9]+([.][0-9]+)*) LTS$"
```
Comment (mwoehlke-kitware) [inline driver/configurations/cache.cmake:399]: If this is optional, a more correct regex would be:
```suggestion
       "^Ubuntu ([0-9]+[.][0-9]+([.][0-9]+)?) LTS$"
```
Or you could just generalize it to support any number of... numbers:
```suggestion
       "^Ubuntu ([0-9]+([.][0-9]+)+) LTS$"
```

File: driver/configurations/cache.cmake [hunk 2]
```diff
@@ -392,9 +392,11 @@ else()
       OUTPUT_VARIABLE DASHBOARD_LSB_RELEASE_DESCRIPTION_OUTPUT_VARIABLE
       OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    # NOTE: Focal reports `Ubuntu 20.04.4 LTS` whereas Jammy reports
+    # `Ubuntu 22.04 LTS`, the third number must be optional.
```
Comment (mwoehlke-kitware) [inline driver/configurations/cache.cmake:396]:
```suggestion
    # `Ubuntu 22.04 LTS`; the third number must be optional.
```

...though this makes me wonder if Focal reported `20.04` at launch, or if Jammy might eventually report `22.04.1`, `22.04.2`, and so forth.

File: driver/configurations/cache.cmake (hunk 1 above)
Comment (svenevs) [reply driver/configurations/cache.cmake]: Went with the second one :+1:

File: driver/configurations/cache.cmake (hunk 2 above)
Comment (svenevs) [reply driver/configurations/cache.cmake]: Comment changed, irrelevant now.

**Comment by svenevs** [comment]
This should be ready to go now, focal builds use the cache, jammy everything builds are successful.

Once nightly and continuous builds start ramping up after this merges we should expect speedups for experimental builds on jammy.

**Review by BetsyMcPhail — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/169#-:-N6SyWw_ASRXSI-nPcOf:bg7igkv)* status: all files reviewed, 2 unresolved discussions (waiting on @mwoehlke-kitware)

**Pull request merged**